tests/property/ic_api/candid/test/tests.ts

```diff
@@ -1,110 +1,78 @@
+            const constraints: DefinitionConstraints = {
+                recursiveWeights: true,
+                weights: {
+                    func: 0, // random func is not supported for didc
+                    null: 0, // null comes out of didc random as (null) and out of candidDecode as (null: null)
+                    opt: 0, // None comes out of didc random as (null) and out of candidDecode as (null: null)
+                    record: 0, // record property names don't get decoded as actual names
+                    variant: 0, // variant property names don't get decoded as actual names
+                    float32: 0, // float32 without a decimal point aren't accepted by candidEncodeQuery
+                    float64: 0, // float64 with a decimal point is accepted by candidEncodeQuery
+                    blob: 0, // blobs are different from didc random and candidDecode, but in a way that seems broken
+                    nat8: 0 // nat8 if paired with vec will make a blob, so we have to filter it out too
+                }
+            };
             await fc.assert(
                 fc.asyncProperty(
-                    fc.uint8Array({
-                        minLength: 10,
-                        maxLength: 100,
-                        min: 32,
-                        max: 126
-                    }),
-                    async (arbitraryBytes) => {
-                        const candidBytes = toCandidBytes(arbitraryBytes);
-
-                        const decodedString = await checkCandidDecoding(
-                            canister,
-                            candidBytes
-                        );
-
-                        const encodedBytes = await checkCandidEncoding(
-                            canister,
-                            arbitraryBytes,
-                            decodedString
-                        );
-
-                        expect(encodedBytes).toEqual(candidBytes);
+                    candidDefinitionArb({ api: 'class', constraints }, {}),
+                    async (candid) => {
+                        const didVisitorResult =
+                            candid.definition.candidMeta.runtimeTypeObject
+                                .getIdlType([])
+                                .accept(
+                                    new DidVisitor(),
+                                    getDefaultVisitorData()
+                                );
+                        const candidString = didVisitorResult[0];
+                        const command = `didc random -t '(${candidString})'`;
+                        const candidValueString = execSyncPretty(command)
+                            .toString()
+                            .trim();
+                        console.info(candidValueString);
+
+                        const canister =
+                            await getCanisterActor<Actor>('canister');
+
+                        const encodedBytes =
```

Comment:
Should we also go the other way?

Reply:
We do on line 67. Unless you mean start with bytes and go to candid. That is VERY difficult to do with arbitrary candid. It was pretty straight forward with just strings, but I thought we were not wanting to do that anymore

Reply:
Comment about doing this once we have the ultimate power arbitrary?

Reply:
done